### DISCUSSION TITLE
Is it possible to add a brush chart to a timeline chart?

### DISCUSSION THREAD

**Discussion by jenniarg21** [apexcharts/apexcharts.js#2753] [Q&A]
I configured a timeline chart with a brush chart but it's not working properly.

Here's what's happening:
- The selection area doesn't apply the min and max ranges, and instead selects the entire range on the brush chart.
- The zoom on the timeline chart does have the min and max ranges applied.
- When modifying the selection area on the brush chart, the timeline chart does not always update to the new min and max ranges. I have to create new selection areas multiple times before the timeline chart applies the new min and max range.

Here's a codepen showing what happens:
https://codepen.io/jenniarg21/pen/abyqwMN

Has anyone implemented this before? Is it not possible to add a brush chart ([docs link](https://apexcharts.com/angular-chart-demos/line-charts/brush-chart/)) to a timeline chart ([docs link](https://apexcharts.com/angular-chart-demos/timeline-charts/advanced/))?

**Comment by VinceUrbanFox** [comment]
Did you ever get this sorted? I'm up against the same thing: keep getting the error: "TypeError: Cannot read properties of null (reading 'niceMin')"

It only works if you draw a new range in the brush chart from left to right.

**Reply by jenniarg21** [reply]
@VinceUrbanFox I don't think it ever got fixed. I ended up using the [jQuery UI range slider](https://jqueryui.com/slider/#range) to allow users to have better control of the date ranges on the timeline chart. My requirements were to use fiscal years rather than calendar years, so the codepen I'm sharing is set up for that. Hope you find it helpful--although, a caveat is that there is a bug with annotations overflowing on the y-axis labels when sliding (the whole reason for setting up the codepen) but it does work. Also, it's good to note that the codepen is not a final solution but a start. I did create a final solution for my clients and they liked it but I left the project over a year ago.

[CodePen: Timeline Chart & ZoomX Event (Apexcharts)](https://codepen.io/jenniarg21/pen/dyzeoMK)

**Reply by VinceUrbanFox** [reply]
Thanks @jenniarg21, that's a decent solution alright, such a shame to have this problem- it almost works and would look so good! Thanks for the codepen: I'll consider either using this approach or trying my hand with D3 to implement something similar. 

@junedchhipa - I'll just tag you here on the off chance that It could bring your awareness to this issue. It may indeed be something trivial to solve within the package and honestly it would be much appreciated, do you think you could kindly take a look? It seems like it may be related to [this](https://github.com/apexcharts/apexcharts.js/issues/3607). Many thanks for any input! :)

**Comment by junedchhipa** [comment]
A PR was recently merged for this issue. 
https://github.com/apexcharts/apexcharts.js/pull/4071
